<robot name="omnidirectional_robot">
    <!-- ベースリンク -->
    <link name="base_link">
        <visual>
            <geometry>
                <cylinder radius="0.15" length="0.02" />
            </geometry>
            <material name="gray" />
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.15" length="0.02" />
            </geometry>
        </collision>
        <inertial>
            <mass value="5.0" />
            <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.2" />
        </inertial>
    </link>

    <!-- wheel -->
    <link name="wheel_1">
        <inertial>
            <mass value="2" />
            <inertia ixx="1e-3" ixy="0.0" ixz="0" iyy="1e-3" iyz="0" izz="1e-3" />
        </inertial>
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.02" />
            </geometry>
            <material name="white" />
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.02" />
            </geometry>
            <contact_coefficients mu="1" kp="1e+13" kd="1.0" />
        </collision>
    </link>
    <joint name="wheel_1_joint" type="continuous">
        <origin xyz="0.15 0 0.0" rpy="1.57 0.0 1.57" />
        <parent link="base_link" />
        <child link="wheel_1" />
        <axis xyz="0.0 0.0 1.0" />
    </joint>
    
    <link name="wheel_2">
        <inertial>
            <mass value="2" />
            <inertia ixx="1e-3" ixy="0.0" ixz="0" iyy="1e-3" iyz="0" izz="1e-3" />
        </inertial>
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.02" />
            </geometry>
            <material name="white" />
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.02" />
            </geometry>
            <contact_coefficients mu="1" kp="1e+13" kd="1.0" />
        </collision>
    </link>
    <joint name="wheel_2_joint" type="continuous">
        <origin xyz="-0.106 0.106 0" rpy="1.57 0.0 0.785398" />
        <parent link="base_link" />
        <child link="wheel_2" />
        <axis xyz="0.0 0.0 1.0" />
    </joint>

    <link name="wheel_3">
        <inertial>
            <mass value="2" />
            <inertia ixx="1e-3" ixy="0.0" ixz="0" iyy="1e-3" iyz="0" izz="1e-3" />
        </inertial>
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.02" />
            </geometry>
            <material name="white" />
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.02" />
            </geometry>
            <contact_coefficients mu="1" kp="1e+13" kd="1.0" />
        </collision>
    </link>
    <joint name="wheel_3_joint" type="continuous">
        <origin xyz="-0.106 -0.106 0" rpy="1.57 0.0 -0.785398" />
        <parent link="base_link" />
        <child link="wheel_3" />
        <axis xyz="0.0 0.0 1.0" />
    </joint>
</robot>